ui/pages/swaps/index.js

```diff
@@ -349,9 +349,12 @@ export default function Swap() {
   }
 
   const redirectToDefaultRoute = async () => {
+    history.push({
+      pathname: DEFAULT_ROUTE,
+      state: { stayOnHomePage: true },
```

Comment:
Is this and relating logic in the homepage component meant to override the state that would result in them being directed back to the swaps route?

Reply:
Yes. Without this a user would go to the homepage from Swaps, but since the Swaps' state wasn't cleaned up yet, the user would be redirected to Swaps again, which we don't want to. However, if a user just clicks in another tab, closes the extension window with Swaps and opens it again, they will be redirected back to Swaps, because the `location.state.stayOnHomePage` param will be falsy.